src/main/java/de/zalando/sprocwrapper/SProcCall.java

```diff
@@ -41,10 +41,35 @@ public long getSprocId() {
     }
 
     public static enum WriteTransaction {
-        USE_FROM_SERVICE,
-        NONE,
-        ONE_PHASE,
-        TWO_PHASE
+        USE_FROM_SERVICE {
+            @Override
+            public SProcService.WriteTransaction getServiceWriteTransaction(SProcService.WriteTransaction serviceWriteTransaction) {
+                if (serviceWriteTransaction == null) {
```

Comment:
I fully understand your intention to simplify the code, but I am still not convinced, that this code belongs to the annotation marker. Maybe it makes sense to have a small static function that does what you want and hides the logic as nicely as you did it now?